monitor/launch/launcher.go

```diff
@@ -96,3 +121,70 @@ func (nm *NodeMonitor) setState(state *models.MonitorStatus) {
 	nm.state = state
 	nm.Mutex.Unlock()
 }
+
+// SendEvent sends an event to the node monitor which will then distribute to
+// all monitor listeners
+func (nm *NodeMonitor) SendEvent(typ int, event interface{}) error {
+	var buf bytes.Buffer
+
+	if err := gob.NewEncoder(&buf).Encode(event); err != nil {
+		nm.bumpLost()
+		return fmt.Errorf("Unable to gob encode: %s", err)
+	}
+
+	select {
+	case nm.queue <- append([]byte{byte(typ)}, buf.Bytes()...):
```

Comment:
Isn't typ already in the event object? In either case, doesn't this break the API we're presenting current clients?

Reply:
Oh, nevermind. This typ is different than the one you replaced.